tests/bson/bson-arraylist-getIterator-001.phpt

```diff
@@ -0,0 +1,145 @@
+--TEST--
+MongoDB\BSON\ArrayList::getIterator tests
+--FILE--
+<?php
+
+$array = [
+    MongoDB\BSON\Document::fromJSON('{ "_id": { "$oid": "56315a7c6118fd1b920270b1" }}'),
```

Comment:
We can take them out of the surrounding document, but then we wouldn't receive actual BSON type instances. This is similar to `MongoDB\BSON\fromJSON` which also won't return type instances if it's in the root document:
```
php > var_dump(MongoDB\BSON\toPHP(MongoDB\BSON\fromJSON('{ "$oid": "56315a7c6118fd1b920270b1" }')));
object(stdClass)#1 (1) {
  ["$oid"]=>
  string(24) "56315a7c6118fd1b920270b1"
}
```

Reply:
The alternative would be to use actual BSON types here (e.g. ObjectId class), no?

Consider the following lines later in the test:

```php
$iterator = MongoDB\BSON\ArrayList::fromPHP($array)->getIterator();

foreach ($iterator as $key => $value) {
    var_dump($key);

    if ($value instanceof MongoDB\BSON\Document || $value instanceof MongoDB\BSON\ArrayList) {
        var_dump(get_class($value));
        var_dump($value->toPHP());
    } else {
        var_dump($value);
    }
}
```

The `else` condition there seems like dead code, because the values are always either a Document or ArrayList.

Reply:
See: https://github.com/mongodb/mongo-php-driver/pull/1387#discussion_r1042228525